library/core/src/any.rs

```diff
@@ -803,111 +831,23 @@ pub const fn type_name_of_val<T: ?Sized>(_val: &T) -> &'static str {
 }
 
 ///////////////////////////////////////////////////////////////////////////////
-// Provider trait
+// Request and its methods
```

Comment:
All this still should be moved out of `any` and into `error`.

Reply:
I wondered about that whenever I first started working on this PR. My reasoning for leaving it here was that it seems more semantically aligned with the content of `any` than `error` and even though we are currently rejecting the `Provider` trait for the sake of moving forward with `error_generic_member_access`, it seemed to me possible that one of the following might happen in the future:

* other standard library types than `Error` might end up using it
* the consensus around the `Provider` trait itself (or something similar) could change

But I'll go ahead and move it assuming you've already thought about these things. Worst case scenario, I suppose the implementation could be moved back into `any` and `error` could continue to provide access to it in its API surface for backwards compatibility's sake.